SUILesson5/SUILesson5/ContentView.swift

```diff
@@ -7,15 +7,205 @@
 
 import SwiftUI
 
+final class ViewModel: ObservableObject {
+
+    @Published var currentCompany = Company(
+        name: Constants.mersedecName,
+        models: [Constants.mercedesAName, Constants.mercedesCName, Constants.mercedesDName]
+    )
+
+    @Published var company = [
+        Company(
+            name: Constants.mersedecName,
+            models: [Constants.mercedesAName, Constants.mercedesCName, Constants.mercedesDName]
+        ),
+        Company(
+            name: Constants.bmwName,
+            models: [Constants.bmw1Name, Constants.bmw3Name, Constants.bmw5Name]
+        ),
+        Company(
+            name: Constants.audiName,
+            models: [Constants.audi1Name, Constants.audi3Name, Constants.audi5Name]
+        ),
+    ]
+
+    @Published var currentLotPrice: Float = 0
+
+    func setCurrentCompany(tag: Int) {
+        currentCompany = company[tag]
+    }
+
+    func getModel(tag: Int) -> String {
+        currentCompany.models[tag]
+    }
+
+    func setCurrentLot(price: Float) {
+        currentLotPrice = price
+    }
+}
+
+struct Company {
+    let name: String
+    let models: [String]
+}
+
 struct ContentView: View {
+
+    @State var segmentIndexCompany = 0
+    @State var segmentIndexCarsModel = 0
+    @State var isBetAlertShown = false
+    @State var isInfoAlertShown = false
+    @State var offsetX = 0
+    @State var priceStape: Float = 0
+    @State var vstackOffset: Float = 0
+    @State var hstackOffset: Float = -250
+    var currentPrice = Int.random(in: 1000000...7000000)
+    @ObservedObject var viewModel = ViewModel()
```

Comment:
всё сделай приватным и ниже свойства тоже сделай приватными
```suggestion
    @ObservedObject var viewModel = ViewModel()
    @State var segmentIndexCompany = 0
    @State var segmentIndexCarsModel = 0
    @State var isBetAlertShown = false
    @State var isInfoAlertShown = false
    @State var offsetX = 0
    @State var priceStape: Float = 0
    @State var vstackOffset: Float = 0
    @State var hstackOffset: Float = -250
    var currentPrice = Int.random(in: 1000000...7000000)

```

Reply:
+ приватные свойства располагай ниже body (https://github.com/ILYA-2606/SwiftUIGuideline)